### ISSUE TITLE
508-defect-2: Review page, form, and results page are inconsistent with each other and design system guidance. They should be reviewed.

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#24644]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** Josh

### Inconsistencies:
- Question grey box in review seems to be a custom component on the review page
- Question grey box in review seems to be a custom component on the review page
- The review page's "Question" title is not consistent with its original source "Additional questions for your providers." 
- "Routine or follow up visit" is an `h2` on the form but becomes a standard response in the review page.
- The styling and code of location, date, and time is inconsistent throughout the full form. Sometimes it's a description list, sometimes it's not. Sometimes it's fully bolded, sometimes it's not.

#### defect-2
- "Additional questions for your provider" should be its own step in the form. It should not be combined with "what's the reason for your visit" (one thing per page) as it is logically not a part of the heading "What's the reason for your visit." This will also help reduce the complexity for Angela's identified launch blocker regarding additional questions.
- [Print and view's accessible label and visible label are different, which will fail to meet label in name](https://www.w3.org/WAI/WCAG21/Techniques/failures/F96.html). Either set the accessible label's name (the `aria-label` to be worded as "Print and view [document name, file type, file size]" OR word the visible label as "Download [document name, file type, file size]"

#### defect-3
- Review page responses are right aligned, but should be left aligned. The jagged left edge makes it difficult for individuals with cognitive or reading disabilities to parse content.
- The results page has two links that are styled as buttons. They should be styled as links.
- "For example" text within the "Are there any other concerns..." is a part of the label itself instead of being separate hint text below the label. Including the "For example" bit as a part of the label and not as hint text (which uses `aria-describedby`) creates an overly verbose label for screen reader users.

#### defect-4
- "Questions" in "Add another question" is capitalized which is inconsistent with button content case style

#### pending questions
- "Routine or follow up visit" being an `h2` does not feel appropriate; the content and IA may need another review. The review page suggests it is something that can be edited, which makes this even more confusing.

## Acceptance Criteria

- [ ] Design is reviewed again for inconsistencies and issues

## Screenshots or Trace Logs

<img width="675" alt="Screen Shot 2021-05-12 at 6 57 32 PM" src="https://user-images.githubusercontent.com/14154792/118054940-305e2380-b355-11eb-8071-c3fb191ab928.png">

<img width="745" alt="Screen Shot 2021-05-12 at 6 57 25 PM" src="https://user-images.githubusercontent.com/14154792/118054926-2a684280-b355-11eb-85eb-d0eed3890e3e.png">

<img width="451" alt="Screen Shot 2021-05-12 at 7 50 03 PM" src="https://user-images.githubusercontent.com/14154792/118058055-48d13c80-b35b-11eb-9cf7-e8ff65143710.png">

<img width="648" alt="Screen Shot 2021-05-12 at 7 49 53 PM" src="https://user-images.githubusercontent.com/14154792/118058058-4a026980-b35b-11eb-9574-4097cfe70162.png">

**Comment by mdewey** [comment]
@kristenmcconnell  I have some questions on this feedback for the design team on how should we remedy these

1. The styling and code of location, date, and time is inconsistent throughout the full form. Sometimes it's a description list, sometimes it's not. Sometimes it's fully bolded, sometimes it's not, how should this be addressed. 

I have other questions but they are around the list loop, so I will hold off on those. 


I will start working on the ones I can and I will create & link a ticket to those.

**Comment by kristenmcconnell** [comment]
We are going to go with the label always being bold and the description not unless @joshkimux has some other feedback. I ask that question, because the completely bold styling was based upon feedback from Jennifer Strickland.

We will also be removing the list loop and replacing it with a text area.

**Comment by joshkimux** [comment]
@kristenmcconnell if it's helpful, the default styling for a description list is to bold the titles (or labels) `dt` and unbold the description `dd`. That's because the titles (or labels) are typically shorter than their ensuing description. All bolded text is more difficult to parse through compared to unbolded text in longer strings.

I'd love to see Jennifer's original feedback or intent if it's there to learn if it was a case by case recommendation or a sweeping one. I definitely could be in the wrong not knowing the full context of the design.

**Comment by mdewey** [comment]
To sum up the ticket status: 

- The inconsistencies have all been addressed
- defect-2s have been addressed
- defect-3s have been documented to be addressed at a later date
- defect-4s is no longer an issue
- For the Pending questions, the "Route or Follow up visit" is no longer an h2. 

Are we good to close this @joshkimux? Do you see anything else?

**Comment by joshkimux** [comment]
@mdewey This is looking great, thank you so much for the fast and quality work.

Some additional items after another review:

## Headings below the stepper need consistency

### 1. `h2` "You have an upcoming appointment" should be a `h3` because it is a hierarchically below the step
<img width="1354" alt="Screen Shot 2021-06-03 at 1 40 52 PM" src="https://user-images.githubusercontent.com/14154792/120688898-849a8600-c471-11eb-90e4-ce5cefa082d9.png">

###  2. This _looks_ like a `h2` but is actually a `p`. I think this is a design and content problem that needs to be discussed. 
<img width="1418" alt="Screen Shot 2021-06-03 at 1 47 16 PM" src="https://user-images.githubusercontent.com/14154792/120689534-38037a80-c472-11eb-9d10-7ff5dcbd60fb.png">

#### Semantically, it would make more sense with cleaner content and as a `h3` that's _not_ contained within a `section`

<img width="1436" alt="Screen Shot 2021-06-03 at 1 50 06 PM" src="https://user-images.githubusercontent.com/14154792/120689912-9d576b80-c472-11eb-8801-c41430cc6b5b.png">

## Description (Definition) lists should not use an `aria-label`
They will be announced as a grouped list, so there's no need to use the `aria-label` to make clear associations.

<img width="1442" alt="Screen Shot 2021-06-03 at 1 44 51 PM" src="https://user-images.githubusercontent.com/14154792/120690504-4bfbac00-c473-11eb-87bf-0fecd980e24b.png">

**Comment by mdewey** [comment]
Hey @joshkimux! All of these are great catches. I think those are big enough to warrant their own ticket (don't want to put too much on this ticket). 

Can you create a new ticket and label the defect levels so I can prioritize these in future sprints?

**Comment by joshkimux** [comment]
@mdewey Sure thing. I've split them into two tickets #25736 and #25739 🚀

**Comment by joshkimux** [comment]
Closing this out as other issues have been resolved 👍
